test/mocha/10-authorization.js

```diff
@@ -0,0 +1,307 @@
+      result.authorization.should.eql(authorization);
+
+      const collection = database.collections['zcap-storage-authorization'];
+      const findResult = await collection.find({
+        controller: helpers.hash(authorization.controller),
+        id: helpers.hash(authorization.capability.id)
+      }).toArray();
+      findResult.should.have.length(1);
+      findResult[0].authorization.should.eql(authorization);
+    });
+    it(`properly inserts an authorization with 'invocationTarget' being an ` +
+      'object', async () => {
+      let err;
+      let result;
+      const authorization = clone(mockData.authorizations.alpha);
+      authorization.capability.invocationTarget = {
+        id: 'urn:uuid:e30d372c-7ab2-429c-91b0-03dc3bcc6289'
+      };
+      try {
+        result = await brZcapStorage.authorizations.insert({
+          controller: authorization.controller,
+          capability: authorization.capability
+        });
+      } catch(e) {
+        err = e;
+      }
+      assertNoError(err);
+      should.exist(result);
+      result.authorization.should.eql(authorization);
+
+      const collection = database.collections['zcap-storage-authorization'];
+      const findResult = await collection.find({
+        controller: helpers.hash(authorization.controller),
+        id: helpers.hash(authorization.capability.id)
+      }).toArray();
+      findResult.should.have.length(1);
+      findResult[0].authorization.should.eql(authorization);
+    });
+    it(`returns DuplicateError on same 'id' and 'controller'`, async () => {
+      const authorization = clone(mockData.authorizations.alpha);
+
+      // insert alpha authorization
+      await brZcapStorage.authorizations.insert({
+        controller: authorization.controller,
+        capability: authorization.capability
+      });
+
+      // attempt to insert same authorization again
+      let err;
+      let result;
+      try {
+        result = await brZcapStorage.authorizations.insert({
+          controller: authorization.controller,
+          capability: authorization.capability
+        });
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(result);
+      should.exist(err);
+      err.name.should.equal('DuplicateError');
+    });
+    it(`returns TypeError when 'invocationTarget.id' is not a string `,
+      async () => {
+        let err;
+        let result;
+        const authorization = clone(mockData.authorizations.alpha);
+        authorization.capability.invocationTarget = {
+          id: {}
+        };
+        try {
+          result = await brZcapStorage.authorizations.insert({
+            controller: authorization.controller,
+            capability: authorization.capability
+          });
+        } catch(e) {
+          err = e;
+        }
+        should.not.exist(result);
+        should.exist(err);
+        err.name.should.equal('TypeError');
+        err.message.should.equal('"capability.invocationTarget" must be a ' +
```

Comment:
I think we should avoid checking English error messages and rely on error types only. We don't want to do any code branching based on English messages anyway.